### PR TITLE
removes fms_io and replaces it fms2_io

### PR DESCRIPTION
**Pull request by uramirez8707**
This just replaces the fms_io calls with fms2_io. 

The FMS ci tries to use this model to compile FMScoupler so once fms_io is removed, it won't work
https://github.com/NOAA-GFDL/FMScoupler/blob/main/t/null_model_build.sh

### CODE REVIEW

**Comment by mcallic2** [comment]
Should I pull the `ocean null` changes into my #ifdef branch and push it or just wait until this is merged?

**Comment by uramirez8707** [comment]
@mcallic2 Once this is merged, they can rerun the CI and it should all pass

**Comment by mcallic2** [comment]
Perf

**Review by nikizadehgfdl — APPROVED** [review]

**Pull request merged**
